node/ledger/assembler_ledger.go

```diff
@@ -141,16 +141,14 @@ func (l *AssemblerLedger) Append(batch types.Batch, orderingInfo types.OrderingI
 	}()
 
 	block := &common.Block{
-		Header: &common.BlockHeader{
-			Number:       ordInfo.Number, // TODO make sure we start from 0
-			DataHash:     ordInfo.Digest,
-			PreviousHash: ordInfo.PrevHash,
-		},
+		Header: ordInfo.CommonBlock.Header,
 		Data: &common.BlockData{
 			Data: batch.Requests(),
 		},
 	}
 
+	block.Header.PreviousHash = ordInfo.PrevHash
```

Comment:
why is this necessary? the `Header: ordInfo.CommonBlock.Header` should already have it, no?